packages/shared/src/hooks/useUserMention.ts

```diff
@@ -203,54 +208,58 @@ export function useUserMention({
       return;
     }
 
-    const [col, row] = getCaretPostition(commentRef.current);
-    replaceWord(commentRef.current, [col + 1, row], '', '@', true);
-
-    commentRef.current.dispatchEvent(
-      new KeyboardEvent('keydown', { key: '@' }),
-    );
+    const textarea = commentRef.current;
+    const start = textarea.selectionStart;
+    const left = textarea.value.substring(0, start);
+    const right = textarea.value.substring(start);
+    textarea.value = `${left} @ ${right}`;
+    textarea.focus();
+    textarea.selectionEnd = start + 2;
+
+    textarea.dispatchEvent(new KeyboardEvent('keydown', { key: '@' }));
   };
 
-  const userClicked = useCallback(
-    (event: MouseEvent & { path: HTMLElement[] }) => {
+  useEffect(() => {
+    const userClicked = (event: MouseEvent) => {
       event.stopPropagation();
       if (typeof query === 'undefined') {
         return;
       }
 
       const el = event.target as HTMLElement;
       const isTextarea = el.tagName.toLocaleLowerCase() === 'textarea';
-      const isTooltip = anyElementClassContains(event.path, 'tippy-content');
-
+      const isTooltip = parentClassContains(el, 'tippy-content');
       if (isTextarea || isTooltip) {
         return;
       }
 
       setQuery(undefined);
-    },
-    [query],
-  );
+    };
 
-  useEffect(() => {
-    document.addEventListener('mousedown', userClicked);
+    const dom = commentRef.current?.getRootNode();
```

Comment:
Just wondering if this doesn't cause issues in the companion?

Reply:
That was actually the reason for this change, as it proves to have some issues when using just "document", so to make it work for the Companion, we need to add the listener to the companion's root node.